sdk/Sdk.Generators/FunctionMetadataProviderGenerator/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -481,35 +481,47 @@ private bool TryGetReturnTypePropertyBindings(ITypeSymbol returnTypeSymbol, bool
                         foundHttpOutput = true;
                         bindingsList.Add(GetHttpReturnBinding(prop.Name));
                     }
-                    else if (prop.GetAttributes().Length > 0) // check if this property has any attributes
+
+                    if (prop.GetAttributes().Length > 0)
                     {
-                        var foundPropertyOutputAttr = false;
+                        var bindingAttributes = prop.GetAttributes().Where(p => p.AttributeClass!.IsOrDerivedFrom(_knownFunctionMetadataTypes.BindingAttribute));
+
+                        if (bindingAttributes.Count() > 1)
+                        {
+                            _context.ReportDiagnostic(Diagnostic.Create(DiagnosticDescriptors.MultipleBindingsGroupedTogether, Location.None, new object[] { "Property", prop.Name.ToString() }));
+                            bindingsList = null;
+                            return false;
+                        }
 
-                        foreach (var attr in prop.GetAttributes()) // now loop through and check if any of the attributes are Binding attributes
+                        // Check if this property has an HttpResultAttribute on it
+                        if (HasHttpResultAttribute(prop))
                         {
-                            if (SymbolEqualityComparer.Default.Equals(attr.AttributeClass?.BaseType, _knownFunctionMetadataTypes.OutputBindingAttribute))
+                            if (foundHttpOutput)
                             {
-                                // validate that there's only one binding attribute per property
-                                if (foundPropertyOutputAttr)
-                                {
-                                    _context.ReportDiagnostic(Diagnostic.Create(DiagnosticDescriptors.MultipleBindingsGroupedTogether, Location.None, new object[] { "Property", prop.Name.ToString() }));
-                                    bindingsList = null;
-                                    return false;
-                                }
+                                _context.ReportDiagnostic(Diagnostic.Create(DiagnosticDescriptors.MultipleHttpResponseTypes, Location.None, new object[] { returnTypeSymbol.Name }));
```

Comment:
The diagnostic error uses "HttpResponseData" in the message, which may not be accurate if someone is applying `HttpResult` attribute on more than one properties.

Reply:
Edited the error, and double checked that this doc was still general enough for the new case: https://github.com/Azure/azure-functions-dotnet-worker/blob/main/docs/analyzer-rules/AZFW0007.md